cluster/gce/gci/configure-helper.sh

```diff
@@ -0,0 +1,750 @@
+# using systemctl. This function assembles the command line and start the kubelet
+# systemd service.
+function start-kubelet {
+  echo "Start kubelet"
+  local flags="${KUBELET_TEST_LOG_LEVEL:-"--v=2"} ${KUBELET_TEST_ARGS:-}"
+  flags+=" --allow-privileged=true"
+  flags+=" --babysit-daemons=true"
+  flags+=" --cgroup-root=/"
+  flags+=" --cloud-provider=gce"
+  flags+=" --cluster-dns=${DNS_SERVER_IP}"
+  flags+=" --cluster-domain=${DNS_DOMAIN}"
+  flags+=" --config=/etc/kubernetes/manifests"
+  flags+=" --kubelet-cgroups=/kubelet"
+  flags+=" --system-cgroups=/system"
+
+  if [[ -n "${KUBELET_PORT:-}" ]]; then
+    flags+=" --port=${KUBELET_PORT}"
+  fi
+  if [[ "${KUBERNETES_MASTER:-}" == "true" ]]; then
+    flags+=" --enable-debugging-handlers=false"
+    flags+=" --hairpin-mode=none"
+    if [[ ! -z "${KUBELET_APISERVER:-}" && ! -z "${KUBELET_CERT:-}" && ! -z "${KUBELET_KEY:-}" ]]; then
+      flags+=" --api-servers=https://${KUBELET_APISERVER}"
+      flags+=" --register-schedulable=false"
+      flags+=" --reconcile-cidr=false"
+      flags+=" --pod-cidr=10.123.45.0/30"
+    else
+      flags+=" --pod-cidr=${MASTER_IP_RANGE}"
+    fi
+  else # For nodes
+    flags+=" --enable-debugging-handlers=true"
+    flags+=" --api-servers=https://${KUBERNETES_MASTER_NAME}"
+    if [[ "${HAIRPIN_MODE:-}" == "promiscuous-bridge" ]] || \
+       [[ "${HAIRPIN_MODE:-}" == "hairpin-veth" ]] || \
+       [[ "${HAIRPIN_MODE:-}" == "none" ]]; then
+      flags+=" --hairpin-mode=${HAIRPIN_MODE}"
+    fi
+  fi
+  if [[ "${ENABLE_MANIFEST_URL:-}" == "true" ]]; then
+    flags+=" --manifest-url=${MANIFEST_URL}"
+    flags+=" --manifest-url-header=${MANIFEST_URL_HEADER}"
+  fi
+  if [[ -n "${ENABLE_CUSTOM_METRICS:-}" ]]; then
+    flags+=" --enable-custom-metrics=${ENABLE_CUSTOM_METRICS}"
+  fi
+  if [[ -n "${NODE_LABELS:-}" ]]; then
+    flags+=" --node-labels=${NODE_LABELS}"
+  fi
+  if [[ "${ALLOCATE_NODE_CIDRS:-}" == "true" ]]; then
+     flags+=" --configure-cbr0=${ALLOCATE_NODE_CIDRS}"
+  fi
+  echo "KUBELET_OPTS=\"${flags}\"" > /etc/default/kubelet
+
+  systemctl start kubelet.service
+}
+
+# Create the log file and set its properties.
+#
+# $1 is the file to create.
+function prepare-log-file {
+  touch $1
+  chmod 644 $1
+  chown root:root $1
+}
+
+# Starts kube-proxy pod.
+function start-kube-proxy {
+  echo "Start kube-proxy pod"
+  prepare-log-file /var/log/kube-proxy.log
+  local -r src_file="${KUBE_HOME}/kube-manifests/kubernetes/kube-proxy.manifest"
+  remove-salt-config-comments "${src_file}"
+
+  local -r kubeconfig="--kubeconfig=/var/lib/kube-proxy/kubeconfig"
+  local kube_docker_registry="gcr.io/google_containers"
+  if [[ -n "${KUBE_DOCKER_REGISTRY:-}" ]]; then
+    kube_docker_registry=${KUBE_DOCKER_REGISTRY}
+  fi
+  local -r kube_proxy_docker_tag=$(cat /home/kubernetes/kube-docker-files/kube-proxy.docker_tag)
+  local api_servers="--master=https://${KUBERNETES_MASTER_NAME}"
+  sed -i -e "s@{{kubeconfig}}@${kubeconfig}@g" ${src_file}
```

Comment:
FWIW, and we can resolve this as a follow-on, it looks like this won't work on OS X (and probably hasn't ever) (BSD `sed` has a different syntax for `-i`). We should just start searching for gnu *everything* like the build script searches for gnu `tar`.